src/commands/login.rs

```diff
@@ -10,9 +13,57 @@ pub struct LoginSubcommand {
     pub token: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct DeviceCodeResponse {
+    device_code: String,
+    user_code: String,
+    verification_uri: String,
+    expires_in: u64,
+    interval: u64,
+}
+
+#[derive(Deserialize)]
+struct DeviceCodeAuth {
+    access_token: String,
+    token_type: String,
+    scope: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum AuthResponse {
+    Ok(DeviceCodeAuth),
+    Err { error: String },
+}
+
+fn await_github_auth(device_code_response: DeviceCodeResponse) -> anyhow::Result<DeviceCodeAuth> {
```

Comment:
Since `await` is a keyword and a known concept, how about `wait_for_github_auth`? :D